File: generate_coin_change.py
coins = [25, 10, 5, 1]

cents = 94

def divide(start_num, divisor):
	quotient = start_num / divisor
	remainder = start_num % divisor

	return quotient, remainder

for x in coins: 
	coin_number, cents = divide(cents, x)
	print 'number of coins:', coin_number, x, 'cent pieces'
	print "remaining cents:", cents
	if cents == 0:
		break





# divide(94, 25)

# divide(19, 10)

# divide(9, 5)

# divide(4, 1)

# Further challenge:
# Push all coin_number values into a list. Then iterate through list, only printing the ones that have value > 0.

# Make a dictionary that has name of coin + value of coin.

# coins_and_values = {
# 	'Quarter' : 25,
# 	'Dime' : 10,
# 	'Nickel' : 5,
# 	'Penny' : 1,
# }

# for k,v in coins_and_values.items():
# 	print k, ":", v

# print coins_and_values['Quarter']	

File: generate_coin.change.js

function generateCoinChange(cents) {
	var quarters = Math.floor(cents / 25);
	var remainder = cents % 25;
	console.log("The change is: " + quarters + " quarter(s)." + "There are now " + remainder + " cents remaining.");

	var dimes = Math.floor(remainder / 10);
	var remainder = remainder % 10;
	console.log("The change is: " + dimes + " dime(s)." + "There are now " + remainder + " cents remaining");

	var nickels = Math.floor(remainder / 5);
	var remainder = remainder % 5;
	console.log("The change is: " + nickels + " nickel(s)." + "There are now " + remainder + " cents remaining.");

	var pennies = Math.floor(remainder / 1);
	console.log("The chanage is: " + pennies + " pennies.");

	}

generateCoinChange(94);





// Challenge: Create a function that accepts a number of cents and outputs how best to represent that amount of money in smallest number of coins (quarters, dimes, nickels, pennies).

// Example:
// 94 cents can be represented as:
// 	quarters: 3
// 	dimes: 1
// 	nickels: 1
// 	pennies: 4 -->

// <!-- Pseudo-code:
// 1. Quarters. Divide argument (94 cents) by 25. Store quotient (3) and remainder (19). Print: Quarters: 3, 19 cents left.
// 2. Dimes. Divide remainder (19) by 10. Store quotient (1) and remainder (9). Print: Dimes: 1, 9 cents left.
// 3. Nickels. Divide remainder (9) by 5. Store quotient (1) and remainder (4). Print: Nickels: 1, 4 cents left.
// 4. Pennies. Divide remainder (4) by 1. Store quotient (4). Print: Pennies: 4. Print: 4 pennies, 0 cents left.
// 5. End program. 